ludwig/hyperopt/execution.py

```diff
@@ -750,12 +775,34 @@ def __init__(
         self.gpu_resources_per_trial = gpu_resources_per_trial
         self.kubernetes_namespace = kubernetes_namespace
         self.time_budget_s = time_budget_s
-
-    def _run_experiment(self, config, checkpoint_dir, hyperopt_dict, decode_ctx):
+        self.max_concurrent_trials = max_concurrent_trials
+        self.sync_config = None
+
+    @property
+    def _cpu_resources_per_trial_non_none(self):
+        return self.cpu_resources_per_trial or 1
+
+    @property
+    def _gpu_resources_per_trial_non_none(self):
+        return self.gpu_resources_per_trial or 0
+
+    def _get_sync_client_and_remote_checkpoint_dir(self, trial_dir: Path):
+        """Get the Ray sync client and path to remote checkpoint directory."""
+        remote_checkpoint_dir = os.path.join(
+            self.sync_config.upload_dir, *_get_relative_checkpoints_dir_parts(trial_dir))
+        return get_cloud_sync_client(remote_checkpoint_dir), remote_checkpoint_dir
+
+    def _validate_remote_fs_for_ray_backend(self, backend, dataset, output_directory):
```

Comment:
As discussed, we should probably remove this. Maybe we can leave it in as a warning. Also, we can use the `split_protocol` approach to determine if it's remote.